### ISSUE TITLE
How to reproduce FID from paper?

### ISSUE THREAD

**Issue by AlexSidDev** [HaozheLiu-ST/T-GATE#12]
Hi! I'am trying to reproduce results of T_GATE (FID metric) that described in your technical report using SDXL model, DPM scheduler with 25 inference steps and gate step is 10. I'am using MS_COCO 256x256 benchmark from https://github.com/Nota-NetsPresso/BK-SDM.git repository and got very big FID instead of 22.738 that presented in your paper on arxiv. Other metrics that I measure like Inception score and CLIP score is normal. Can you please provide more information about hyperparameters (guidance scale for example), image resolution? What captions used for generation (full validation set from MSCOCO-2014 or MSCOCO-2017, or maybe some subset from them) and what real images was used to measure FID between real and generated samples?

**Comment by HaozheLiu-ST** [comment]
YES. The index is generated by [this dataloader](https://huggingface.co/datasets/ydshieh/coco_dataset_script). And FID is based on [pytorch_fid](https://github.com/mseitzer/pytorch-fid). 
If you find this project useful for you, please consider giving a star and citation if available.

**Comment by HaozheLiu-ST** [comment]
Hello, 

after re-checking the code, we find that we incorporate some random operations in preparing data. 

To reproduce our results, we will upload a caption and image index later. @AlexSidDev @WentianZhang-ML 

Best,
Haozhe

**Comment by WentianZhang-ML** [comment]
Hi @AlexSidDev, sorry for the late reply. 

We re-checked the code and re-generated images to reproduce FID. You can use the attached file ([idx_caption.txt](https://github.com/HaozheLiu-ST/T-GATE/files/15369063/idx_caption.txt)) to compute FID, which contains 10,000  randomly selected image IDs and their corresponding captions.

Here are the reproduced FID results:

|            | FID    |
|------------|--------|
| SDXL       | 24.164 |
| SDXL (m=5)  | 22.109 |
| SDXL (m=10) | 22.917 |



> Hello,
> 
> after re-checking the code, we find that we incorporate some random operations in preparing data.
> 
> To reproduce our results, we will upload a caption and image index later. @AlexSidDev @WentianZhang-ML
> 
> Best, Haozhe

**Comment by AlexSidDev** [comment]
Hi, @WentianZhang-ML. Thank you very much.
